contracts/src/v0.8/shared/token/ERC20/BurnMintERC20.sol

```diff
@@ -0,0 +1,205 @@
+    return i_decimals;
+  }
+
+  /// @dev Returns the max supply of the token, 0 if unlimited.
+  function maxSupply() public view virtual returns (uint256) {
+    return i_maxSupply;
+  }
+
+  /// @dev Uses OZ ERC20 _transfer to disallow sending to address(0).
+  /// @dev Disallows sending to address(this)
+  function _transfer(address from, address to, uint256 amount) internal virtual override validAddress(to) {
+    super._transfer(from, to, amount);
+  }
+
+  /// @dev Uses OZ ERC20 _approve to disallow approving for address(0).
+  /// @dev Disallows approving for address(this)
+  function _approve(address owner, address spender, uint256 amount) internal virtual override validAddress(spender) {
+    super._approve(owner, spender, amount);
+  }
+
+  /// @dev Exists to be backwards compatible with the older naming convention.
+  /// @param spender the account being approved to spend on the users' behalf.
+  /// @param subtractedValue the amount being removed from the approval.
+  /// @return success Bool to return if the approval was successfully decreased.
+  function decreaseApproval(address spender, uint256 subtractedValue) external returns (bool success) {
+    return decreaseAllowance(spender, subtractedValue);
+  }
+
+  /// @dev Exists to be backwards compatible with the older naming convention.
+  /// @param spender the account being approved to spend on the users' behalf.
+  /// @param addedValue the amount being added to the approval.
+  function increaseApproval(address spender, uint256 addedValue) external {
+    increaseAllowance(spender, addedValue);
+  }
+
+  // ================================================================
+  // │                      Burning & minting                       │
+  // ================================================================
+
+  /// @inheritdoc ERC20Burnable
+  /// @dev Uses OZ ERC20 _burn to disallow burning from address(0).
+  /// @dev Decreases the total supply.
+  function burn(uint256 amount) public override(IBurnMintERC20, ERC20Burnable) onlyBurner {
+    super.burn(amount);
+  }
+
+  /// @inheritdoc IBurnMintERC20
+  /// @dev Alias for BurnFrom for compatibility with the older naming convention.
+  /// @dev Uses burnFrom for all validation & logic.
+  function burn(address account, uint256 amount) public virtual override {
+    burnFrom(account, amount);
+  }
+
+  /// @inheritdoc ERC20Burnable
+  /// @dev Uses OZ ERC20 _burn to disallow burning from address(0).
+  /// @dev Decreases the total supply.
+  function burnFrom(address account, uint256 amount) public override(IBurnMintERC20, ERC20Burnable) onlyBurner {
+    super.burnFrom(account, amount);
+  }
+
+  /// @inheritdoc IBurnMintERC20
+  /// @dev Uses OZ ERC20 _mint to disallow minting to address(0).
+  /// @dev Disallows minting to address(this)
+  /// @dev Increases the total supply.
+  function mint(address account, uint256 amount) external override onlyMinter validAddress(account) {
```

Comment:
Do we need to disallow minting to self? Feels pretty normal

Reply:
that's just how it is in the other tokens we've done I'm guessing as some form of re-entrancy but since we're not doing a 677 that seems reasonable to remove the check.

Reply:
My bad, this is disallowing minting to the token address, not to the msg.sender